### PR TITLE
Do not mark a result as good if it is also bad

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14268

---

A result can only be "good" if it is not "bad".

### CODE REVIEW

File: src/olympia/scanners/tests/test_views.py [hunk 1]
```diff
@@ -326,3 +332,29 @@ def test_get_results_with_blocked_versions(self):
...
+        )
+        VersionLog.objects.create(
+            activity_log=ActivityLog.create(
+                action=amo.LOG.CONFIRM_AUTO_APPROVED,
+                version=version_1,
+                user=self.user,
+            ),
+            version=version_1,
+        )
+        # Oh noes! The version has been blocked.
```
Comment (bobsilverberg) [inline src/olympia/scanners/tests/test_views.py:351]: Would it be worth adding a check assertion here to verify that at this point the result would be labelled `good`, to make sure that the reason it is labelled `bad` below is valid?

File: src/olympia/scanners/views.py [hunk 2]
```diff
@@ -46,6 +46,13 @@ def get_queryset(self):
             bad_results = bad_results.filter(scanner=scanner)
             good_results = good_results.filter(scanner=scanner)
 
+        bad_filters = Q(state=TRUE_POSITIVE) | Q(
+            version__versionlog__activity_log__action__in=(
+                amo.LOG.BLOCKLIST_BLOCK_ADDED.id,
+                amo.LOG.BLOCKLIST_BLOCK_EDITED.id,
```
Comment (bobsilverberg) [inline src/olympia/scanners/views.py:52]: I notice that there are two different reasons that a `good` result could be labelled `bad` as a result of being blocked (i.e., `BLOCKLIST_BLOCK_ADDED` and `BLOCKLIST_BLOCK_EDITED`), but it looks like a test was only added for one of these. Is that a concern, or do you feel the test cases are sufficient to cover the code change?

Comment (willdurand) [reply src/olympia/scanners/views.py:52]: I think it is fine as it is.

File: src/olympia/scanners/tests/test_views.py (hunk 1 above)
Comment (willdurand) [reply src/olympia/scanners/tests/test_views.py:351]: It is covered by other test cases so I would say it is fine?

File: src/olympia/scanners/views.py (hunk 2 above)
Comment (bobsilverberg) [reply src/olympia/scanners/views.py:52]: Yes, we do generally take that approach, although I must say I'm not a huge fan of it. It does cut down on test code, but I actually think it's better if one can read a single test in isolation and be able to tell that it's correctly testing what needs to be tested. This approach, which, again, I know we commonly take, means that we have to have knowledge of other tests in the file to be sure that our test setup is correct.

Personally, I think one additional check assertion in a test adds more value than it takes away (due to maintenance), but I know this isn't how we usually do things, and this PR isn't necessarily the place to change that approach. In other words, let's leave it as is, but perhaps this is a topic worth discussing further.

File: src/olympia/scanners/tests/test_views.py (hunk 1 above)
Comment (bobsilverberg) [reply src/olympia/scanners/tests/test_views.py:351]: Yes, we do generally take that approach, although I must say I'm not a huge fan of it. It does cut down on test code, but I actually think it's better if one can read a single test in isolation and be able to tell that it's correctly testing what needs to be tested. This approach, which, again, I know we commonly take, means that we have to have knowledge of other tests in the file to be sure that our test setup is correct.

Personally, I think one additional check assertion in a test adds more value than it takes away (due to maintenance), but I know this isn't how we usually do things, and this PR isn't necessarily the place to change that approach. In other words, let's leave it as is, but perhaps this is a topic worth discussing further.

File: src/olympia/scanners/views.py (hunk 2 above)
Comment (bobsilverberg) [reply src/olympia/scanners/views.py:52]: Oops, I added my reply to the wrong comment. I moved it.

**Pull request merged**